### ISSUE TITLE
sg: support 8 KiB RAM adapters

### ISSUE THREAD

**Issue by invertego** [ares-emulator/ares#56]
Some Taiwanese ports of MSX games required a special adapter containing extra RAM to be played on an SG-1000. [See here for details.](https://www.smspower.org/forums/13579-Taiwan8KBRAMAdapterForPlayingMSXPortsOnSG1000II)

Games known to require these adapters are documented in the [MAME software list](https://github.com/mamedev/mame/blob/master/hash/sg1000.xml).

Type A adapter:
```
<rom name="bomberman special [dahjee] (tw).bin" size="49152" crc="ce5648c3" sha1="455838571dd8287d491431e03269b0ccd292c8b8" offset="000000" />
<rom name="yie ar kung-fu ii [dahjee] (tw).bin" size="49152" crc="fc87463c" sha1="44fb2dca3774f859f0ae0ad4d809a67f760cf9ae" offset="000000" />
<rom name="king's valley (tw).bin" size="32768" crc="223397a1" sha1="922d23895c12707b2fc382ea5cb769c5cb9b5755" offset="000000" />
<rom name="knightmare [jumbo] (tw).bin" size="49152" crc="281d2888" sha1="babdb6b109b20e81c28cd892c2042723e9cbff49" offset="000000" />
<rom name="pippols [jumbo] (tw).sg" size="32768" crc="df7cbfa5" sha1="4a95b98e127fb0c6428d6debcfc92152069ac531" offset="000000" />
<rom name="rally-x [dahjee] (tw).bin" size="32768" crc="306d5f78" sha1="21770691191b62bafec9099bbe0f9942f5480f83" offset="000000" />
<rom name="road fighter [jumbo] (tw).bin" size="32768" crc="29e047cc" sha1="4f1a139974e3db27af5973bc9572ed68acd6ea68" offset="000000" />
<rom name="twinbee [jumbo] (tw).bin" size="49152" crc="c550b4f0" sha1="a8c30043c145e63f79a3265dde167529f0e08d49" offset="000000" />
<rom name="tank battalion [dahjee] (tw).bin" size="32768" crc="5cbd1163" sha1="0ca3b740b28266a2a606127ed8967c684b67cb44" offset="000000" />
<rom name="legend of kage, the [dahjee] (tw).bin" size="49152" crc="2e7166d5" sha1="902dbb122bf0efc76604a876c3fae51c074bdc6a" offset="000000" />
<rom name="star soldier [dahjee] (tw).sg" size="49152" crc="e0816bb7" sha1="c1984257809ec5bce80a65d22b9b32561e04922c" offset="000000" />
```

Type B adapter:
```
<rom name="bomberman special (tw).bin" size="49152" crc="69fc1494" sha1="d1d56bcd996df94458f5901dc537a5e30021256b" offset="000000" />
<rom name="castle, the [msx] (tw).bin" size="49152" crc="2e366ccf" sha1="5a81d5103487695cabd811aabd9e71f48b9e7996" offset="000000" />
<rom name="magical kid wiz (tw).bin" size="49152" crc="ffc4ee3f" sha1="fd24b501046cf811b04386129be0362a1eabf6b3" offset="000000" />
<rom name="rally-x (tw).bin" size="32768" crc="aaac12cf" sha1="0c1957a5faae5254c69f3331e79d87499dd7cf48" offset="000000" />
<rom name="road fighter (tw).bin" size="32768" crc="d2edd329" sha1="2755f74019dc94559fd0c2248e2ecb7f48879b90" offset="000000" />
```

**Comment by LukeUsher** [comment]
as an interesting side-note, the memory map of Type-B actually matches up with the standard Sega Master System; so loading the Type-B games in the 'Master System' core should allow them to work for now.

**Comment by LukeUsher** [comment]
fixed as of 6b6b602
